test/units/galaxy/test_api.py

```diff
@@ -53,6 +55,21 @@ def collection_artifact(tmp_path_factory):
     yield tar_path
 
 
+@pytest.fixture()
+def cache_dir(tmp_path_factory, monkeypatch):
+    cache_dir = to_text(tmp_path_factory.mktemp('Test ÅÑŚÌβŁÈ Galaxy Cache'))
+    orig_get_config = C.config.get_config_value
+
+    def get_config_value(option):
+        overrides = {
+            'GALAXY_CACHE_DIR': cache_dir,
+        }
+        return overrides.get(option, orig_get_config(option))
+    monkeypatch.setattr(C.config, 'get_config_value', get_config_value)
```

Comment:
If I read the source correctly, this can be simplified to
```suggestion

    monkeypatch.setitem(C.config._base_defs, 'GALAXY_CACHE_DIR', cache_dir)
```

Reply:
Doesn't work because the value need to be a dict that contains the actual config definition like

```python
monkeypatch.setitem(C.config._base_defs, 'GALAXY_CACHE_DIR', {'default': cache_dir})
```

Still I'm just going to go the original route as it doesn't rely on `_` attributes and I'm not sure if setting the `default` value like this will still be overridable by environment values.